lib/get-deleted-user-ids-to_local-file.rb

```diff
@@ -0,0 +1,54 @@
+# Usage - call script with bundle exec ruby lib/get-all-user-account-to_local-file-ok.rb | tee /Users/davidpye/alphagov/zendesk-scripts/data/users-to-delete.jsonl
+# above file path used in deletion step.
+
+require 'zendesk_api'
+require 'rest-client'
+require 'json'
+
+@client = ZendeskAPI::Client.new do |config|
+  config.url = ENV['ZENDESK_URL']
+  config.username = ENV['ZENDESK_USER_EMAIL']
+  config.password = ENV['ZENDESK_USER_PASSWORD']
+  config.retry = true
+end
+
+
+# TBC CAN WE LIMIT THE DATE RANGE AS 8000+ PAGES CURRENTLY!!!
+
+begin
+  url = "https://govuk.zendesk.com/api/v2/deleted_users.json"
+  output_results = []
+  # while url != "https://govuk.zendesk.com/api/v2/deleted_users.json?page=5" ### This is fine for testing ###
+  while url do
+    search_results = JSON.parse(RestClient::Request.execute method: :get, url: url, user: ENV['ZENDESK_USER_EMAIL'], password: ENV['ZENDESK_USER_PASSWORD'])
+    (output_results << search_results['deleted_users']).flatten!
+    puts search_results['next_page']
+    # puts "output_results: #{output_results}"
+    url = search_results['next_page']
+  end
+end
+
+File.open("data/soft_deleted_users.json", "w") { |file| file.write(JSON.generate(output_results)) }
+
+
+
+# The Zendesk API has 100 items per page, so programatically
```

Comment:
These comments should be removed. If we need this info, we should add it to documentation somewhere.

Reply:
This was mine, when I did [this](https://github.com/alphagov/zendesk-scripts/pull/5)! I'm glad some of it lives on. I think.

Reply:
This was mine, when I did [this](https://github.com/alphagov/zendesk-scripts/pull/5)! I'm glad some of it lives on. I think.